### PR TITLE
Panel search improvements

### LINKED ISSUE #6409: Search in panel shows old results when response times from server vary
## Description
The search in the panel can show partial/old results when the response time from the server can vary. Example: You want to search for "test". You type slowly, so a first query is launched for "te", then another one when you finished typing "test". If the request for "test" is faster to load, it shows those results. If afterwards the request for "te" completes, it overwrites the results shown. So you seem to see the results for "test" but you actually see the results for "te".

**Expected behavior**  
When a new search query is launched, the old running queries should be canceled so that this behaviour can't happen.

## To reproduce
Easiest way might be to overwrite the search method to include a random delay:
```php
Kirby::plugin('searchtest', [
    // This is overwriting the search used in the panel
    'areas' => [
        'site' => function ($kirby) {
            return [
                'searches' => [
                    'pages' => [
                        'query' => function (string $query = null, int $limit, int $page) use ($kirby) {
                            sleep(rand(0, 3));
                            // Simply pass on to the core search
                            return $kirby->core()->area('site')['searches']['pages']['query']($query, $limit, $page);
                        }
                    ]
                ]
            ];
        },
    ]
]);
```
Then go to the panel and quickly type part of the query, wait half a second, then type the rest. You might have to try a few times so that the first query actually takes longer than the second one.

## Your setup
**Kirby Version**  
4.2.0

**Additional context**  
The behaviour might not happen with just the default setup and few results. But if you overwrite the panel search with a custom search that e.g. fetches data from a big DB or another API, this happens and there's no way to fix it within the custom search.

### PR DESCRIPTION
**Pull request by distantnative**
### Summary of changes
- New `k-search-bar` component to separate UI logic from API interaction
- New `panel.searcher` module
- Ongoing Panel search request get aborted when new one is started
- Search view: add UI to differentiate searching states

### Reasoning
- A "dumber" component that only takes care of the UI logic makes it easier to test the different states with Lab examples.

### Additional context
For testing `k-search`: https://sandbox.test/panel/lab/components/search

### Fixes
- Fixed parallel Panel search requests
https://github.com/getkirby/kirby/issues/6409

### For review team

- [x] Add lab and/or sandbox examples (wherever helpful)
- [x] Add changes & docs to release notes draft in Notion

### CODE REVIEW

File: panel/src/panel/panel.js [hunk 1]
```diff
@@ -72,6 +73,7 @@ export default {
 		this.activation = Activation(this);
 		this.drag = Drag(this);
 		this.events = Events(this);
+		this.searcher = Search(this);
```
Comment (distantnative) [inline panel/src/panel/panel.js:76]: `Ideas?` for a better name than `panel.searcher`? `panel.search` is blocked by the method name

File: panel/src/components/Dialogs/SearchDialog.vue [hunk 2]
```diff
@@ -182,52 +65,8 @@ export default {
 .k-search-dialog {
 	--dialog-padding: 0;
 	--dialog-rounded: var(--rounded);
-	overflow: visible;
```
Comment (distantnative) [inline panel/src/components/Dialogs/SearchDialog.vue]: `Question:` Anyone recall what this was for? With it the border-radius has no effect. Removing it, I din't spot any problems.

File: panel/src/panel/panel.js (hunk 1 above)
Comment (afbora) [reply panel/src/panel/panel.js:76]: finder?

File: panel/src/components/Dialogs/SearchDialog.vue (hunk 2 above)
Comment (bastianallgeier) [reply panel/src/components/Dialogs/SearchDialog.vue]: I think it was about the outline for the input/dropdown button.

**Review by bastianallgeier — COMMENTED** [review]
I really like the abstraction. I would vote for either choosing k-search and k-search-bar though. Right now, the component is called k-search but the class is k-search-bar. I think I'm leaning more towards k-search-bar, but I'm open for both.

**Comment by distantnative** [comment]
@bastianallgeier I guess that's a leftover from when I had it as `k-search-bar`. I am quite indifferent, so let me change it back to `k-search-bar`.

**Review by bastianallgeier — APPROVED** [review]
I renamed the lab example page as well and fixed a small issue with the lab example CSS. But other than that it's a great update.

**Pull request merged**